tensorboard/summary/_tf/summary/__init__.py

```diff
@@ -12,7 +12,73 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
-"""TensorFlow component package for providing tf.summary from TensorBoard."""
+
+# NOTE: This module exists to provide the `tf.summary` module in the TensorFlow
+# API using symbols defined in TensorBoard. This works via a mechanism within
+# TensorFlow's API construction logic called "component_api_helper" that imports
+# an arbitrary module and inserts it into the TF APIs as a "component API". That
+# logic is specifically hardcoded to look for this exact tensorboard module.
+#
+# This note is in a comment, not the module docstring, because the module
+# docstring below is what users will see as the tf.summary docstring and in the
+# generated API documentation, and this is just an implementation detail.
+
+"""
+Operations for writing summary data, for use in analysis and visualization.
+
+The tf.summary module provides APIs for writing summary data. This data can be
+visualized in TensorBoard, the visualization toolkit that comes with TensorFlow.
+See the [TensorBoard website](https://www.tensorflow.org/tensorboard) for more
+detailed tutorials about how to use these APIs, or some quick examples below.
+
+Example usage with eager execution, the default in TF 2.0:
+
+```python
+writer = tf.summary.create_file_writer("/tmp/mylogs")
+with writer.as_default():
+  for step in range(100):
+    # other model code would go here
+    tf.summary.scalar("metric", 0.5, step=step)
+    writer.flush()
+```
+
+Example usage with `tf.function` graph execution:
+
+```python
+writer = tf.summary.create_file_writer("/tmp/mylogs")
+
+@tf.function
+def my_func(metric, step):
+  # other model code would go here
+  with writer.as_default():
+    tf.summary.scalar("metric", metric, step=step)
+
+for step in range(100):
+  my_func(0.5, step)
```

Comment:
Do we expect it to be common that these metrics are _inputs_ to the
training loop rather than computed inside `my_func`? It’s more
believable in the eager example where `0.5` appears as a literal in the
summary call, but for `metric` to actually be a parameter to `my_func`
seems strange.

Reply:
Yeah, that's a good point, it's not really realistic.  I wanted to show function argument passing but `step` accomplishes that well enough on its own.

Reply:
Fine with me; no big deal.